src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -4395,6 +4395,126 @@ public static void DerivedTypeWithDifferentOverrides()
     }
 #endif
 
+    //This case currently will fail in ReflectionOnly implementation when targeting uapaot
+#if !uapaot || !ReflectionOnly 
+    [Fact]
+    public static void Xml_DefaultNamespaceChange_SimpleArray_ObjectAsRoot()
+    {
+        SimpleType[] x = new SimpleType[] { new SimpleType { P1 = "abc", P2 = 11 }, new SimpleType { P1 = "def", P2 = 12 } };
+#if uapaot && !ReflectionOnly
+        Func<XmlSerializer> serializerFactory = () => { return new XmlSerializer(typeof(object)); };
+#else
+        Func<XmlSerializer> serializerFactory = () => { return new XmlSerializer(typeof(object), new Type[] { typeof(SimpleType[]) }); };
+#endif
+
+        SimpleType[] y = SerializeAndDeserialize(x,
+@"<?xml version=""1.0""?>
+<anyType d1p1:type=""ArrayOfSimpleType"" xmlns:d1p1=""http://www.w3.org/2001/XMLSchema-instance"" >
+  <SimpleType xmlns:xsd=""http://www.w3.org/2001/XMLSchema"">
+    <P1>abc</P1>
+    <P2>11</P2>
+  </SimpleType>
+  <SimpleType>
+    <P1>def</P1>
+    <P2>12</P2>
+  </SimpleType>
+</anyType>", serializerFactory);
+
+        Utils.Equal(x, y, (a, b) => { return SimpleType.AreEqual(a, b); });
+    }
+#endif
+
+    [Fact]
+    public static void Xml_ValidateExceptionOnUnspecifiedRootSerializationType()
+    {
+        var value = new UnspecifiedRootSerializationType();
+        var actual = SerializeAndDeserialize(value, "<?xml version=\"1.0\"?>\r\n<UnspecifiedRootSerializationType xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\">\r\n  <MyIntProperty>0</MyIntProperty>\r\n</UnspecifiedRootSerializationType>", () => { return new XmlSerializer(Type.GetType(typeof(UnspecifiedRootSerializationType).FullName)); });
+        Assert.StrictEqual(value.MyIntProperty, actual.MyIntProperty);
+        Assert.StrictEqual(value.MyStringProperty, actual.MyStringProperty);     
+    }
+
+    [Fact]
+    public static void Xml_VerifyCompilationIssueOnly()
+    {
+        MemoryStream ms = new MemoryStream();
+        Assert.Throws<InvalidOperationException>(() => 
```

Comment:
You can refactor this into a method as shown below,

```c#
        private static void AssertSerializationFailure<T, ExceptionType>()
        {
            using (var ms = new MemoryStream())
            {
                Assert.Throws<ExceptionType>(() =>
                {
                    var serializer = new XmlSerializer(typeof(T));
                    var value = new T();
                    serializer.Serialize(ms, value);
                });
            }
        }  
```
You can then use the method for the types below. For example,
```
AssertSerializationFailure<TypeWithEnumerableMembers, InvalidOperationException>();
```